src/diffusers/pipelines/wuerstchen/modules.py

```diff
@@ -0,0 +1,389 @@
+import math
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from diffusers.configuration_utils import ConfigMixin, register_to_config
+
+from ...models.modeling_utils import ModelMixin
+
+
+class LayerNorm2d(nn.LayerNorm):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def forward(self, x):
+        return super().forward(x.permute(0, 2, 3, 1)).permute(0, 3, 1, 2)
+
+
+class TimestepBlock(nn.Module):
+    def __init__(self, c, c_timestep):
+        super().__init__()
+        self.mapper = nn.Linear(c_timestep, c * 2)
+
+    def forward(self, x, t):
+        a, b = self.mapper(t)[:, :, None, None].chunk(2, dim=1)
+        return x * (1 + a) + b
+
+
+class Attention2D(nn.Module):
```

Comment:
We should not define a new attention module here - instead we should take: https://github.com/huggingface/diffusers/blob/71c82241593fbfbbc30f783d9f034dbadb91df0d/src/diffusers/models/attention_processor.py#L36